lightningnetwork/lnd
# lnd: use persisted node announcement data from previous runtime

closes #7123 

replaces #8690 

## Change Description
Description of change / link to associated issue.
In this commit, the node announcement config is retrieved from the database and reused when a node restarts.

We will begin by checking the configuration set in lnd.conf  file or passed as arguments when the node starts. If the configuration is not present, we will check for the ones saved on disk and use them. If none are found, we will resort to using the default configurations.

## Steps to Test
Steps for reviewers to follow to test the change.
- Make sure no alias or color is set in your lnd.conf (config here takes precedence)
- Use the `updatenodeannouncement` RPC call to update your node alias, color, addresses or feature bits. 
- Verify the alias is set using lncli getinfo
- Now stop and start the node again and run lncli getinfo again
You'll notice that the fields you update earlier are persisted


## Pull Request Checklist
### Testing


### Code Style and Documentation

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.
